core/templates/dev/head/pages/exploration_editor/translation_tab/StateTranslationDirective.js

```diff
@@ -290,16 +290,8 @@ oppia.directive('stateTranslation', [
             $scope.answerChoices = StateEditorService.getAnswerChoices(
               $scope.stateInteractionId, currentCustomizationArgs);
             $scope.onTabClick($scope.TAB_ID_CONTENT);
+            $scope.$broadcast('refreshAudioTranslationBar');
           };
-
-          // TODO(DubeySandeep): We need to call initStateTranslation() here in
-          // case the listener that receives 'refreshStateTranslation' is not
-          // set by the time it is broadcasted from ExplorationEditor.js. Figure
-          // out a solution that does not rely on covering the race condition.
-          if (ExplorationStatesService.isInitialized()) {
-            $scope.initStateTranslation();
-          }
-          $rootScope.loadingMessage = '';
```

Comment:
Moved this to AudioTranslation Bar such that page gets view once each dom is loaded.